apps-rendering/src/components/NewsletterSignup/EmailSignupForm.tsx

```diff
@@ -50,50 +127,94 @@ const EmailSignupForm: FC<Props> = ({ newsletterId, format }) => {
 					}
 				`}
 			/>
-			<form css={formStyle} action={undefined}>
-				<TextInput
-					type="email"
-					width={30}
-					hideLabel
-					label="Enter your email address"
-					cssOverrides={css`
-						height: 2.25rem;
-						margin-right: ${remSpace[3]};
-						margin-top: 0;
-						margin-bottom: ${remSpace[2]};
-						flex-basis: ${pxToRem(335)}rem;
+			<form
+				className={'js-signup-form'}
+				data-newsletter-id={newsletterId}
+				css={formStyle}
+			>
+				<div
+					className={'js-signup-form__inputs'}
+					css={css`
+						align-items: center;
+						flex-wrap: wrap;
+					`}
+				>
+					<TextInput
+						type="email"
+						width={30}
+						hideLabel
+						label="Enter your email address"
+						cssOverrides={css`
+							height: 2.25rem;
+							margin-right: ${remSpace[3]};
+							margin-top: 0;
+							margin-bottom: ${remSpace[2]};
+							flex-basis: ${pxToRem(335)}rem;
 
-						${darkModeCss`
+							${darkModeCss`
 							background-color: ${background.signUpFormDark(format)};
 							color: ${text.signUpFormDark(format)};
 						`}
-					`}
-				/>
-				<Button
-					onClick={handleSubmit}
-					size="small"
-					title="Sign up"
-					cssOverrides={css`
-						background-color: ${fill.signUpFormButton(format)};
-						color: ${text.signUpFormButton(format)};
-						margin-bottom: ${remSpace[2]};
-						flex-basis: ${pxToRem(118)}rem;
-						justify-content: center;
-						&:hover {
-							background-color: ${hover.signUpFormButton(format)};
-						}
-
-						${darkModeCss`
-							background-color: ${fill.signUpFormButtonDark(format)};
-							color: ${text.signUpFormButtonDark(format)};
-							&:hover {
-								background-color: ${hover.signUpFormButtonDark(format)}
-							}
 						`}
+					/>
+					<Button
+						onClick={handleSubmit}
+						size="small"
+						title="Sign up"
+						type="submit"
+						cssOverrides={buttonStyle(format)}
+					>
+						Sign up
+						<span className="js-signup-form__feedback js-signup-form__feedback--waiting">
+							<SvgSpinner size="small" />
```

Comment:
if we used React for interactivity, we could use `isLoading` on `Button` instead